bot/cogs/grades_cog.py

```diff
@@ -0,0 +1,259 @@
+            except Exception as e:
+                print(e)
+            
+            if data[0] > bestProfAB:
+                bestProfName = i
+                bestProfAB = data[0]
+                bestProfLenCount = data[-1]
+            if data[1] > worstProfFW:
+                worstProfName = i
+                worstProfFW = data[1]
+                worstProfLenCount = data[-1]
+        
+        profString = "The statistically best professor is " + bestProfName + " with an A+B Avg of " + str(bestProfAB) + "% in " + str(bestProfLenCount) + " classes\n\nThe statistically worst professor is " + worstProfName + " with an F+W Avg of " + str(worstProfFW) + "% out of " + str(worstProfLenCount) + " class(es)" #Final professor string
+
+        return courseString + "\n\n" + profString + "\n\n"
+
+    def getInitials(self, Name):
+        initials = ""
+        Name = Name.split()
+        Name = Name[1:] + [Name[0]] #Rotates it
+        for i in Name:
+            initials += i[0]
+        return initials
+
+    def initialize(self):
+
+        if os.path.isfile('bot/cogs/Student-Teacher/assets/master.json'): #SKIP PARSE DATA WHEN NOT NECESSARY
+            print("FOUND COURSE FILE")
+            with open('bot/cogs/Student-Teacher/assets/master.json', 'r') as f:
+                self.master_list = json.load(f)
+            # print(master_list)
+            if os.path.isfile('bot/cogs/Student-Teacher/assets/master_prof.json'):
+                print("FOUND PROF FILE")
+                with open('bot/cogs/Student-Teacher/assets/master_prof.json', 'r') as f:
+                    self.master_prof_list = json.load(f)
+            else:
+                print("PROF FILE NOT FOUND")
+        else:
+            print("FILE NOT FOUND PLEASE SEND HELP")
+            
+            
+
+    def getFirstLast(self, Name):
+        FML = Name.split()
+        First = FML[1]
+        Last = FML[0]
+        return First + " " + Last
+
+    def process_profQuery(self, Name):
+        
+        data_list = self.master_prof_list[Name]
+        
+        A = []
+        B = []
+        F = []
+        W = []
+        P = []
+        NP = []    
+
+        C = []
+        D = [] # Incase you want these 
+
+        for i in data_list:
+            A.append(i['A'])
+            B.append(i['B'])
+            C.append(i['C'])
+            D.append(i['D'])
+            F.append(i['F'])
+            W.append(i['W'])
+        gradesList = (A,B,C,D,F,W,P,NP)
+        for letterGrade in gradesList:
+            for i in range(len(letterGrade)):
+                letterGrade[i] = int(letterGrade[i][:-1])
+
+        AvgA = sum(A)//len(A)
+        AvgB = sum(B)//len(B)
+        AvgC = sum(C)//len(C)
+        AvgD = sum(D)//len(D)
+        AvgF = sum(F)//len(F)
+        AvgWithdraw = sum(W)//len(W)
+
+        return  [(AvgA + AvgB), (AvgF+AvgWithdraw), (len(data_list))]
+
+    def toCamelCase(self, Name):
+        items = Name.split()
+        for i in range(len(items)):
```

Comment:
This would be much clearer and more pythonic with the use of enumerate() instead of range(Len())